src/Provider/AbstractProvider.php

```diff
@@ -356,12 +356,7 @@ protected function fetchProviderData($url, array $headers = [])
         try {
             $client = $this->getHttpClient();
             $client->setBaseUrl($url);
```

Comment:
I honestly don't understand why our tests are asserting for the number of times a method is called internally. I think that makes for brittle tests, and we should probably just remove those lines.